### PR TITLE
fix(wallet): Bridge Transaction Parsing

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 

Parses `Bridge` transaction info correctly on the `Confirm Transaction` panel and `Activity` tab

Resolves <https://github.com/brave/brave-browser/issues/38552>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Open the `Wallet` and navigate to the `Bridge` screen
2. Select some tokens to bridge and get a quote
3. Click the `Review Bridge` button and check that information correct in the `Confirm Transaction` panel
4. Navigate to the `Activity` tab and make sure information is displayed correctly for that transaction

Screenshot 46

Screenshot 47

Screenshot 48

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/popup-modals/transaction_details_modal/transaction_details_modal.tsx [hunk 1]
```diff
@@ -181,6 +184,16 @@ export const TransactionDetailsModal = ({ onClose, transaction }: Props) => {
     chainId: transaction.chainId,
     coin: txCoinType
   })
+
+  const { data: toNetwork } = useGetNetworkQuery(
+    isBridgeTx && transaction.swapInfo?.toChainId && transaction.swapInfo.toCoin
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/popup-modals/transaction_details_modal/transaction_details_modal.tsx:189]: `swapInfo.toCoin` could possibly be `0` which is a valid option but is "falsey"

File: components/brave_wallet_ui/components/desktop/portfolio_transaction_item/portfolio_transaction_item.tsx [hunk 2]
```diff
@@ -144,6 +158,15 @@ export const PortfolioTransactionItem = React.forwardRef<HTMLDivElement, Props>(
       coin: txCoinType
     })
 
+    const { data: toNetwork } = useGetNetworkQuery(
+      isBridge && transaction.swapInfo?.toChainId && transaction.swapInfo.toCoin
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/portfolio_transaction_item/portfolio_transaction_item.tsx:162]: `toCoin` can be `0` (falsey)

File: components/brave_wallet_ui/components/desktop/popup-modals/transaction_details_modal/transaction_details_modal.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/transaction_details_modal/transaction_details_modal.tsx]: Good catch! Now checking for `undefined`.

File: components/brave_wallet_ui/components/desktop/portfolio_transaction_item/portfolio_transaction_item.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/portfolio_transaction_item/portfolio_transaction_item.tsx]: Now checking for `undefined`.

**Review by josheleonard — APPROVED** [review]
LGTM

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.70.60 Chromium: 128.0.6613.18 (Official Build) nightly (64-bit)
-- | --
Revision | f110df81ade140122b1c2a8cf40f7d25ef15b1bc
OS | Linux

- Verified steps from PR description
- Verified Bridge transaction parsing works as expected

https://github.com/user-attachments/assets/98920117-2550-4d75-9a95-880bc8dff315

**Pull request merged**